plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/core/GlobalRuntimeEnv.java

```diff
@@ -8,25 +8,29 @@
 
 package org.opensearch.datafusion.core;
 
-import static org.opensearch.datafusion.DataFusionQueryJNI.closeGlobalRuntime;
-import static org.opensearch.datafusion.DataFusionQueryJNI.createGlobalRuntime;
-import static org.opensearch.datafusion.DataFusionQueryJNI.createTokioRuntime;
+import org.opensearch.datafusion.jni.NativeBridge;
+
+import java.lang.ref.Cleaner;
 
 /**
  * Global runtime environment for DataFusion operations.
- * Manages the lifecycle of the native DataFusion runtime.
+ * Manages the lifecycle of the native DataFusion runtime with automatic cleanup.
  */
-public class GlobalRuntimeEnv implements AutoCloseable {
-    // ptr to runtime environment in df
+public final class GlobalRuntimeEnv implements AutoCloseable {
+
+    private static final Cleaner CLEANER = Cleaner.create();
+
     private final long ptr;
-    private final long tokio_runtime_ptr;
+    private final long tokioRuntimePtr;
+    private final Cleaner.Cleanable cleanable;
 
     /**
      * Creates a new global runtime environment.
      */
     public GlobalRuntimeEnv() {
-        this.ptr = createGlobalRuntime();
-        this.tokio_runtime_ptr = createTokioRuntime();
+        this.ptr = NativeBridge.createGlobalRuntime();
```

Comment:
How are we thinking to handle the error handling? In the rust side we are returning back 0, but there is no cache here for it.